src/Epic/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { css, ThemeProvider } from '@emotion/react';
+import React from 'react';
+import { brand, palette, space } from '@guardian/src-foundations';
+import { OphanComponentEvent } from '@guardian/types';
+import { BrazeClickHandler } from '../utils/tracking';
+import { ContributionsEpicButtons } from './ContributionsEpicButtons';
+import { body, headline } from '@guardian/src-foundations/typography';
+import { replaceNonArticleCountPlaceholders } from './placeholders';
+import { canRenderEpic, parseParagraphs } from './canRender';
+
+// Custom styles for <a> tags in the Epic content
+const linkStyles = css`
+    a {
+        color: ${palette.news[400]};
+        text-decoration: none;
+    }
+    a:hover {
+        text-decoration: underline;
+    }
+`;
+
+const styles = {
+    epicWrapper: css`
+        max-width: 620px;
+        margin: 10px;
+    `,
+    epicContainer: css`
+        padding: 4px 8px 12px;
+        border-top: 1px solid #ffe500;
+        background-color: #f6f6f6;
+        display: flex;
+        flex-direction: column;
+    `,
+    paragraph: css`
+        margin: 0 auto ${space[2]}px;
+        ${body.medium()}
+        ${linkStyles}
+    `,
+    heading: css`
+        ${headline.xxsmall({ fontWeight: 'bold' })}
+        margin-top: 0;
+        margin-bottom: ${space[3]}px;
+    `,
+    highlightText: css`
+        ${body.medium({ fontWeight: 'bold' })}
+        ${linkStyles}
+        padding: 2px;
+        background-color: ${palette.brandAlt[400]};
+    `,
+};
+
+export type BrazeMessageProps = {
+    buttonText?: string;
+    buttonUrl?: string;
+    heading?: string;
+    highlightedText?: string;
+    ophanComponentId?: string;
+    paragraph1?: string;
+    paragraph2?: string;
+    paragraph3?: string;
+    paragraph4?: string;
+    paragraph5?: string;
+    paragraph6?: string;
+    paragraph7?: string;
+    paragraph8?: string;
+    paragraph9?: string;
+};
+
+export type EpicProps = {
+    logButtonClickWithBraze: BrazeClickHandler;
+    submitComponentEvent: (componentEvent: OphanComponentEvent) => void;
+    ophanComponentId?: string;
+    brazeMessageProps: BrazeMessageProps;
+    countryCode?: string;
+};
+
+export const Epic: React.FC<EpicProps> = (props: EpicProps) => {
+    const {
+        // logButtonClickWithBraze,
+        // submitComponentEvent,
+        // ophanComponentId = COMPONENT_NAME,
+        brazeMessageProps: { heading, buttonText, buttonUrl, highlightedText },
+        countryCode,
+    } = props;
+
+    if (!canRenderEpic(props.brazeMessageProps)) {
+        return null;
+    }
+
+    const paragraphs = parseParagraphs(props.brazeMessageProps).map((paragraph) =>
+        replaceNonArticleCountPlaceholders(paragraph, countryCode),
+    );
+
+    const highlightTextClean = replaceNonArticleCountPlaceholders(highlightedText, countryCode);
```

Comment:
There is a code path in `replaceNonArticleCountPlaceholders` where we fail to replace `%%COUNTRY_NAME%%` because we can't derive the country name from the country code passed. Which means there _could_ be unreplaced placeholders here. I wonder how important that is.

It'd be interesting to know which (if any) or these placeholders might be useful (or already used) by marketing. Pretty sure country name isn't currently.

We could return `null` early here, like the contributions epic does. But if we do that I think we'd want to add the check to our `canRender`, but that increases the complexity of `canRender` quite a bit as we'd need to have access to the country code, and try to do the replacement in there, which seems overly complex.

What do you reckon? Maybe it would be better to do any replacement like this using liquid template logic in Braze, that way we wouldn't have to support placeholders at all (we should check first if we're already using them for existing braze epic campaigns).

Reply:
I think I agree that using Liquid logic is the right way to go here. It does seem silly to build our own version considering there's nothing that we support that can't be done via Liquid.

Reply:
Yeah I guess the dealbreaker will be the data we want to use to use to drive the templating. For example the country Braze knows about (which is what we'd have to use in the Liquid) is potentially different than the country in the browser which comes from Fastly's geolocation service.